Authority check in SecurityUtils support array parameter

iconben opened the pull request:
To provide some handy methods in scenario:   
- want to check if a user has any authority in a list of authorities
- want to check if a user has none of authority in a list of authorities

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

<!--
Please also reference the issue number in a commit message to [automatically close the related GitHub issue](https://help.github.com/articles/closing-issues-via-commit-messages/)

Note: It is also possible to add `[skip ci]` or `[ci skip]` to your commit message to skip continuous integration tests
-->

iconben commented:
Hi @mshima, I saw some checks failed due to network error, what should I do?

The error:
```
Could not download Saxon-HE-9.9.1-7.jar (net.sf.saxon:Saxon-HE:9.9.1-7)
      > Could not get resource 'https://plugins.gradle.org/m2/net/sf/saxon/Saxon-HE/9.9.1-7/Saxon-HE-9.9.1-7.jar'.
```

vishal423 commented:
Since these are utility methods and are currently not used anywhere in code, I suggest moving this to server-side library code in  https://github.com/jhipster/jhipster-bom project.

iconben commented:
> Since these are utility methods and are currently not used anywhere in code, I suggest moving this to server-side library code in https://github.com/jhipster/jhipster-bom project.

Should we merge this first, then discuss whether the whole SecurityUtils class should be moved to jhipster-bom ?

vishal423 commented:
I will rather close this PR as it would be surplus code for applications. JHipster server-side library is the perfect place to add these utility methods. Also, don't forget to add appropriate Junit tests.

pascalgrimaud commented:
Not sure about that @vishal423 

The current SecurityUtils is already generated by generator-jhipster and is used to check this kind of thing.
For my current customer, I added some specific method like this one, because the generated one is not enough.

So the good place is here. Anyway, this will add some dead code as it won't be used any where. Am I correct @iconben ?

iconben commented:
> Not sure about that @vishal423 
> 
> The current SecurityUtils is already generated by generator-jhipster and is used to check this kind of thing.
> For my current customer, I added some specific method like this one, because the generated one is not enough.
> 
> So the good place is here. Anyway, this will add some dead code as it won't be used any where. Am I correct @iconben ?

Yes and no, I added tests as well so they are used... in tests.😀

mshima commented:
IMO just point `hasCurrentUserThisAuthority` to `hasCurrentUserAnyOfAuthorities`.

While `hasCurrentUserNoneOfAuthorities` can be replaced with `!hasCurrentUserAnyOfAuthorities`.

The 3 implementations can reuse a single one.

DanielFran commented on generators/server/templates/src/main/java/package/security/SecurityUtils.java.ejs:
```diff
@@ -140,27 +144,52 @@ public final class SecurityUtils {
+     * @return true if the current user has any of the authorities, false otherwise.
      */
-    public static <% if (!reactive) { %>boolean<% } else { %>Mono<Boolean><% } %> hasCurrentUserThisAuthority(String authority) {
+    public static <% if (!reactive) { %>boolean<% } else { %>Mono<Boolean><% } %>  hasCurrentUserAnyOfAuthorities(String... authorities) {
```
```suggestion
    public static <% if (!reactive) { %>boolean<% } else { %>Mono<Boolean><% } %> hasCurrentUserAnyOfAuthorities(String... authorities) {
```

DanielFran commented on generators/server/templates/src/main/java/package/security/SecurityUtils.java.ejs:
```diff
@@ -140,27 +144,52 @@ public final class SecurityUtils {
+     * @param authorities the authorities to check.
+     * @return true if the current user has none of the authorities, false otherwise.
+     */
+    public static <% if (!reactive) { %>boolean<% } else { %>Mono<Boolean><% } %>  hasCurrentUserNoneOfAuthorities(String... authorities) {
```
```suggestion
    public static <% if (!reactive) { %>boolean<% } else { %>Mono<Boolean><% } %> hasCurrentUserNoneOfAuthorities(String... authorities) {
```

DanielFran commented:
Thanks @iconben for the contribution
